e2e/cypress/support/fetch_commands.js

```diff
@@ -19,3 +19,42 @@ const addDelay = (win, routes, delay) => {
         return fetch(...args);
     });
 };
+
+// # Wrap websocket to be able to connect and close connections on demand
+Cypress.Commands.add('mockWebsockets', (socketList) => {
+    cy.on('window:before:load', (win) => mockWebsockets(win, socketList));
+});
+
+const mockWebsockets = (win, socketList) => {
+    const RealWebSocket = WebSocket;
+    cy.stub(win, 'WebSocket').callsFake((...args) => {
+        const mockWebSocket = {
+            wrappedSocket: null,
+            onopen: null,
+            onmessage: null,
+            onerror: null,
+            onclose: null,
+            send(data) {
+                if (this.wrappedSocket) {
+                    this.wrappedSocket.send(data);
+                } else {
+                    onerror();
+                }
+            },
+            close() {
+                if (this.wrappedSocket) {
+                    this.wrappedSocket.close(1000);
+                }
+            },
+            connect() {
+                this.wrappedSocket = new RealWebSocket(...args);
+                this.wrappedSocket.onopen = this.onopen;
+                this.wrappedSocket.onmessage = this.onmessage;
+                this.wrappedSocket.onerror = this.onerror;
+                this.wrappedSocket.onclose = this.onclose;
+            },
+        };
+        socketList.push(mockWebSocket);
```

Comment:
I'm fine as is though it's a bit weird since we're not normally mutating input.

Reply:
It is not the most elegant way to do it, but due to Cypress singularities, I have not found a more elegant solution. Nevertheless, if you have any proposal, I am more than willing to try it.

Reply:
Yeah, I don't have good recommendation either.  One thing I can think of is by introducing a member in the global `window`, something like:
```javascript
window.mockWebsockets = [];
...
window.mockWebsockets.push(mockWebSocket);

// then use directly
window.mockWebsockets.forEach((websocket) => {
    websocket.connect();
});
```
I think we'll be using this more in the future by injecting on initial Cypress run (connect WS by default), then close/connect mock websockets whenever needed anywhere in the tests.